module/os/linux/spl/spl-thread.c

```diff
@@ -158,3 +158,63 @@ spl_kthread_create(int (*func)(void *), void *data, const char namefmt[], ...)
 	} while (1);
 }
 EXPORT_SYMBOL(spl_kthread_create);
+
+/*
+ * The "why" argument indicates the allowable side-effects of the call:
+ *
+ * FORREAL:  Extract the next pending signal from p_sig into p_cursig;
+ * stop the process if a stop has been requested or if a traced signal
+ * is pending.
+ *
+ * JUSTLOOKING:  Don't stop the process, just indicate whether or not
+ * a signal might be pending (FORREAL is needed to tell for sure).
+ */
+int
+issig(int why)
+{
+	ASSERT(why == FORREAL || why == JUSTLOOKING);
+
+	if (!signal_pending(current))
+		return (0);
+
+	if (why != FORREAL)
+		return (1);
+
+	struct task_struct *task = current;
+#ifdef HAVE_SIGINFO
+	kernel_siginfo_t __info;
+#else
+	siginfo_t __info;
+#endif
+	sigset_t set;
+	siginitsetinv(&set, 1ULL << (SIGSTOP - 1) | 1ULL << (SIGTSTP - 1));
+	sigorsets(&set, &task->blocked, &set);
+
+	spin_lock_irq(&task->sighand->siglock);
+	int ret;
+	if ((ret = dequeue_signal(task, &set, &__info)) != 0) {
+#ifdef HAVE_SIGNAL_STOP
+		spin_unlock_irq(&task->sighand->siglock);
+		kernel_signal_stop();
+#else
+		if (current->jobctl & JOBCTL_STOP_DEQUEUED) {
+#ifdef HAVE_SET_SPECIAL_STATE
+			set_special_state(TASK_STOPPED);
+#else
+			__set_current_state(TASK_STOPPED);
+#endif
```

Comment:
nit: maybe here as well with a `spl_set_special_state()` wrapper.  Though since this is such a small function and it's only needed here I don't think it's a big deal either way.